lib/account.js

```diff
@@ -162,6 +162,101 @@ module.exports = function(config) {
           callback(err, json);
         });
       });
+    },
+    requestMigrateEmail: function(request, callback) {
+      var appURLObj = url.parse(config.uri + '/migrate', true);
+      appURLObj.query = request.payload.oauth;
+
+      var migrateRequest = hyperquest({
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          'x-ratelimit-ip': getIPAddress(request)
+        },
+        uri: loginAPI + '/api/v2/user/request'
+      });
+
+      migrateRequest.on('error', callback);
+
+      migrateRequest.on('response', function(message) {
+        parseMessage(message, function(err, json) {
+          callback(err, json);
+        });
+      });
+
+      migrateRequest.end(JSON.stringify({
+        uid: request.payload.username,
+        appURL: url.format(appURLObj),
+        migrateUser: true
+      }), 'utf8');
+    },
+    verifyToken: function(request, callback) {
+      var verifyRequest = hyperquest({
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          'x-ratelimit-ip': getIPAddress(request)
+        },
+        uri: loginAPI + '/api/v2/user/authenticateToken'
+      });
+
+      verifyRequest.on('error', callback);
+
+      verifyRequest.on('response', function(message) {
+        parseMessage(message, function(err, json) {
+          callback(err, json);
+        });
+      });
+
+      verifyRequest.end(JSON.stringify({
+        uid: request.pre.username,
+        token: request.payload.token
+      }), 'utf8');
+    },
+    setPassword: function(request, callback) {
+      var passwordRequest = hyperquest({
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          'x-ratelimit-ip': getIPAddress(request)
+        },
+        uri: loginAPI + '/api/v2/user/enable-passwords'
+      });
+
+      passwordRequest.on('error', callback);
+
+      passwordRequest.on('response', function(message) {
+        parseMessage(message, function(err, json) {
+          callback(err, json);
+        });
+      });
+
+      passwordRequest.end(JSON.stringify({
```

Comment:
So, when this happens, is there something on `loginAPI` that checks whether or not we should really allow password-setting here?